lib/_tls_wrap.js

```diff
@@ -275,12 +274,10 @@ function TLSSocket(socket, options) {
 
   // Wrap plain JS Stream into StreamWrap
   var wrap;
-  if (!(socket instanceof net.Socket) && socket instanceof Duplex)
-    wrap = new StreamWrap(socket);
-  else if ((socket instanceof net.Socket) && !socket._handle)
-    wrap = new StreamWrap(socket);
-  else
+  if (socket instanceof net.Socket && socket._handle || !socket)
```

Comment:
@addaleax oops, didn't understand it correctly. Sorry.
 I've added parentheses around && expression. Thank you :)